src/mds/Server.cc

```diff
@@ -6253,6 +6261,57 @@ void Server::handle_client_removexattr(MDRequestRef& mdr)
   journal_and_reply(mdr, cur, 0, le, new C_MDS_inode_update_finish(this, mdr, cur));
 }
 
+void Server::handle_client_getvxattr(MDRequestRef& mdr)
+{
+  const cref_t<MClientRequest> &req = mdr->client_request;
+  string name(req->get_path2());
+
+  // is a ceph virtual xattr?
+  if (!is_ceph_vxattr(name)) {
+    respond_to_request(mdr, -CEPHFS_EINVAL);
+    return;
+  }
+
+  // can't use rdlock_path_pin_ref because we need to xlock snaplock/policylock
```

Comment:
I realize this is copy paste but you don't need to xlock for reading vxattrs.

Reply:
ack